pkg/reconciler/domainmapping/reconciler.go

```
@@ -317,6 +372,129 @@ func (r *Reconciler) resolveRef(ctx context.Context, dm *v1beta1.DomainMapping)
+
+// targetTrafficPaths returns the cluster-local traffic paths for a referenced
+// Knative Service or Route. Other Addressables keep the existing single-Service
+// backend behavior.
+func (r *Reconciler) targetTrafficPaths(dm *v1beta1.DomainMapping, targetHost string) ([]netv1alpha1.HTTPIngressPath, bool, error) {
+	refKind, ok := servingReferenceKind(dm.Spec.Ref)
+	if !ok {
+		return nil, false, nil
+	}
+
+	if r.tracker == nil {
+		return nil, true, fmt.Errorf("DomainMapping target tracker is not configured")
```

golangci-lint (2.12.2)

[error] error-format: fmt.Errorf can be replaced with errors.New

(perfsprint)